webboot/webboot.go

```diff
@@ -23,9 +27,62 @@ var (
 	ipv6    = flag.Bool("ipv6", true, "use IPV6")
 )
 
+func wget(arg string) (io.ReadCloser, error) {
+	resp, err := http.Get(arg)
+	if err != nil {
+		return nil, err
+	}
+	if resp.StatusCode != 200 {
+		return nil, fmt.Errorf("non-200 HTTP status: %d", resp.StatusCode)
+	}
+	return resp.Body, nil
+}
+
+func writer(read io.ReadCloser, fileName string) (string, error) {
+	w, err := os.Create(fileName)
+	if err != nil {
+		return "", err
+	}
+
+	defer w.Close()
+
+	_, err = io.Copy(w, read)
+	return fileName, err
+}
+
+func usage() {
+	log.Printf("Only one argument: an ISO link")
+	log.Printf("Usage: %s [ARGS] URL\n", os.Args[0])
+	flag.PrintDefaults()
+	os.Exit(2)
+}
+
 func main() {
 	flag.Parse()
 
+	if flag.NArg() != 1 {
+		usage()
+	}
+
+	argURL := flag.Arg(0)
```

Comment:
I don't think you need this test. There are all kinds of bad URLs they can put in, so just let it get caught by the wget below.